packages/core/core/src/BundleGraph.js

```diff
@@ -808,42 +802,52 @@ export default class BundleGraph {
         if (result.symbol != undefined) {
           if (assetOutside) {
             // We found the symbol, but `asset` is outside, return `asset` and the original symbol
+            maybeFoundInDependencies = true;
             break;
           }
 
           return {
             asset: result.asset,
             symbol: result.symbol,
             exportSymbol: symbol,
-            loc: resolved.symbols.get(symbol)?.loc,
+            loc: resolved.symbols?.get(symbol)?.loc,
           };
         }
+        if (!result.asset.symbols) {
+          // We didn't find it in this dependency, but it might still be there: bailout.
+          maybeFoundInDependencies = true;
+          break;
+        }
       }
     }
 
     return {
       asset,
       exportSymbol: symbol,
-      symbol: identifier,
-      loc: asset.symbols.get(symbol)?.loc,
+      symbol: identifier ?? (maybeFoundInDependencies ? null : undefined),
```

Comment:
Generally I find relying on having a semantic difference between `null` and `undefined` values makes things unclear. Could we use another special value instead? Or is this difference documented somewhere?

Reply:
I agree that this is not ideal, but I don't know of a better way.
I've documented it here
https://github.com/parcel-bundler/parcel/blob/f520e215405a62de59ec34a7d089618663b9cb7e/packages/core/core/src/BundleGraph.js#L737-L740
I'll move it to `@parcel/types`

Reply:
Could we use a sentinel value like `const FOUND_IN_EXPORTS = {}` and use that as the symbol?

Reply:
You mean for `resolveSymbol(...).symbol`?
I would prefer an  ES6 `Symbol` or `false` in that case. Though `void | false | String` does play well with Flow's sketchy-null-string check.